source/_posts/Meta-Learning-Learning-to-learn-fast.md

```diff
@@ -0,0 +1,101 @@
+<!-- # Meta-Learning: Learning to learn fast -->
+
+> In this blog, I will explain meta-learning what is also known as learning to learn in machine learning.
+
+## What is Meta-Learning?
+
+Current AI agents excel at mastering a particular skill like object classification, predicting a disaster, etc. However, now AI is extensively used in day-to-day life. Almost every organization now uses AI in their products. So there is a need for a single AI agent that can solve a variety of problems. ***A good machine learning model often requires training with a large number of samples. Humans, in contrast, learn new concepts and skills much faster and more efficiently.*** Kids who have seen cats and birds only a few times can quickly tell them apart. Similar to humans, machine learning models need to be versatile and adapt to a new problem with the least number of samples. This essentially **meta-learning** aims to solve.
+
+Versatility is achieved by an intelligent amalgamation of meta-learning along with related techniques such as **reinforcement learning** (finding suitable actions to maximize a reward), **transfer learning**    (re-purposing a trained model for a specific task on a second related task), and **active learning** (learning algorithm chooses the data it wants to learn from). Such different learning techniques provide an AI agent with the brains to do multiple tasks without the need to learn every new task from scratch. Thereby making it capable of adapting intelligently to a wide variety of new, unseen situations.
+> **Using Meta-Learning, AI agents learn how to learn new tasks by reusing prior experience, rather than examining each new task in isolation.**
+
+## **Approaches for Meta-Learning**
+
+There are two popular approaches to meta-learning.
+
+### **The Simple View**
+
+A good meta-learning model should be trained over a variety of learning tasks and optimized for the best performance on the distribution of tasks, including potentially unseen tasks. Each task is associated with a dataset D, containing both feature vectors and true labels. The optimal model parameters are:
+
+{% asset_img formula_1.png %}
+<!-- ![](https://cdn-images-1.medium.com/max/2000/1*QjXE5hevXgaSfkIYr5yxdw.png) -->
+
+here one dataset is considered as one sample data point.
+
+### **Few-shot classification**
+
+*Few-shot classification* is an instantiation of meta-learning in the field of supervised learning. The dataset ***D*** is often split into two parts, support set ***S*** for learning and a prediction set ***B*** for training or testing, ***D=*(*S, B*)**. Often we consider a *K-shot N-class classification* task: the support set contains K labeled examples for each of N classes.
+
+{% asset_img image.png  An example of a 4-shot 2-class image classification %}
+
+<!-- ![*An example of a 4-shot 2-class image classification.*](https://cdn-images-1.medium.com/max/2000/1*NzBnn1dqwh_4_eH9vYVj4w.png)**An example of a 4-shot 2-class image classification.** -->
```

Comment:
Optimize your images with image alt text

Reply:
optimized images and added description for images. few images are formulas,so did not add Alt for them